### ISSUE TITLE
`unnecessary_parenthesis` false positive on map literal at the beginning of a line with more than one method call

### ISSUE THREAD

**Issue by DetachHead** [dart-lang/linter#3474]
**Describe the issue**
when calling two methods on a map literal, `unnecessary_parenthesis` is incorrectly reported. when applying the quick fix, it removes the parentheses and causes a syntax error.

**To Reproduce**
```dart
void main() {
  ({'asdf': 'asdf'}).toString().toString();
}
```
![image](https://user-images.githubusercontent.com/57028336/176164314-fbb9898c-73e7-406d-b07f-bbe6c910f0f5.png)
![image](https://user-images.githubusercontent.com/57028336/176164355-00708e5c-d1c3-4bc4-ba3c-179151fb891d.png)

**Expected behavior**
no error as the parentheses here are required
